docker/CMSRucioClient/scripts/updateDDMQuota

```diff
@@ -90,28 +81,35 @@ def calculate_ddm_quotas():
         if static == 0:
             continue  # Skip if static is 0
 
+        # Apply penalty for special rses
+        rse_attributes = client.list_rse_attributes(rse)
+        if "ddm_quota_penalty" in rse_attributes:
+            static, rucio, expired = (x * float(rse_attributes["ddm_quota_penalty"]) for x in (static, rucio, expired))
+
         free = static - rucio
-        ddm_quota = STATIC_WEIGHT * (static / total_static) + FREE_WEIGHT * \
-            (free / static) + EXPIRED_WEIGHT * (expired / static)
-        if MAKE_QUADRATIC:
+        ddm_quota = static_weight * (static / total_static) + free_weight * \
+            (free / static) + expired_weight * (expired / static)
+        if make_quadratic:
             ddm_quota = ddm_quota ** 2
 
         # Override ddm_quota for operational purposes
-        rse_attributes = client.list_rse_attributes(rse)
         if "override_ddm_quota" in rse_attributes:
             overridden_ddm_quotas[rse] = rse_attributes["override_ddm_quota"]
             continue
 
         ddm_quotas[rse] = ddm_quota
+    return ddm_quotas, overridden_ddm_quotas
 
 
-def normalize_ddm_quotas():
+def normalize_ddm_quotas(ddm_quotas):
     weights = [value for value in ddm_quotas.values()]
     for rse, weight in ddm_quotas.items():
         ddm_quotas[rse] = int(((weight - min(weights)) / (max(weights) - min(weights))) * MAX_DDM_QUOTA)
```

Comment:
I think we should discuss the normalisation. This way of doing it sets the ddm_quota for the rse with the smallest value to 0. 
This should not be the case and should be done manually, with the `override_ddm_quota` option.

For disk, it was kinda ok, not optimal, with these numbers changing at a faster pace. For tapes, we would just end up creating another PIC situation.

Reply:
I'd argue that we wouldn't end up in a PIC situation in the long term since as other sites fill up, the RSE with the lowest ddm_quota would change as per the algorithm. Said that, it's also easy to normalize starting from a positive integer

Reply:
Yeah, but the time to change is relatively slow and I don't see why we should randomly not use one rse at all.
So, let's keep the 0 reserved for overrides in exceptional situations. 

And we can do the norm as `x/sum` (or something else) instead of `(x-min)/(max-min)`.